org.adempiere.base/src/org/compiere/model/MDepositBatch.java

```diff
@@ -344,5 +397,248 @@ public boolean isComplete()
+			return DocAction.STATUS_Invalid;
+		}
+		
+		if (log.isLoggable(Level.INFO)) log.info("completeIt - " + toString());
+		
+		MDepositBatchLine[] depositbatchLines = getLines();
+		//	Close lines
+		for (int line = 0; line < depositbatchLines.length; line++)
+		{
+			depositbatchLines[line].setProcessed(true);
+			depositbatchLines[line].saveEx();
+		}
+		
+		//Re-calculate lines total amount and Update Header Deposit Amount
+		updateHeaderAmt();
+		
+		//	User Validation
+		String valid = ModelValidationEngine.get().fireDocValidate(this, ModelValidator.TIMING_AFTER_COMPLETE);
+		if (valid != null)
+		{
+			m_processMsg = valid;
+			return DocAction.STATUS_Invalid;
+		}
+			
+		setProcessed(true);
+		
+		setDocAction(DOCACTION_Close);
+		return DocAction.STATUS_Completed;
+	}
+
+
+	@Override
+	public boolean closeIt() {
+		if (log.isLoggable(Level.INFO)) log.info("closeIt - " + toString());
+		// Before Close
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_CLOSE);
+		if (m_processMsg != null)
+			return false;		
+
+		setDocAction(DOCACTION_None);
+
+		// After Close
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_AFTER_CLOSE);
+		if (m_processMsg != null)
+			return false;
+		return true;
+	}
+
+
+	@Override
+	public boolean reverseCorrectIt() {
+		if (log.isLoggable(Level.INFO)) log.info("reverseCorrectIt - " + toString());
+		// Before reverseCorrect
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_REVERSECORRECT);
+		if (m_processMsg != null)
+			return false;
+		
+		// After reverseCorrect
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_AFTER_REVERSECORRECT);
+		if (m_processMsg != null)
+			return false;
+		return false;
+	}
+
+
+	@Override
+	public boolean reverseAccrualIt() {
+		if (log.isLoggable(Level.INFO)) log.info("reverseAccrualIt - " + toString());
+		// Before reverseAccrual
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_REVERSEACCRUAL);
+		if (m_processMsg != null)
+			return false;
+		
+		// After reverseAccrual
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_AFTER_REVERSEACCRUAL);
+		if (m_processMsg != null)
+			return false;
+		return false;
+	}
+
+
+	@Override
+	public boolean reActivateIt() {
+		if (log.isLoggable(Level.INFO)) log.info("reActivateIt - " + toString());
+		// Before reActivate
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_REACTIVATE);
+		if (m_processMsg != null)
+			return false;		
+		
+		if (log.isLoggable(Level.INFO)) log.info("ReactivateIt - " + toString());
+		// Set Payment reconciled false
+		MDepositBatchLine[] depositbatchLines = getLines();
+
+		// Close lines

```

Comment:
wrong comment

Reply:
Fixed

Reply:
@dpansheriya, the comment is still **Close lines** and the code below is actually doing reactivation of lines.

Reply:
This is fixed